File: order/src/main/java/com/example/order/controller/OrderController.java
package com.example.order.controller;

import com.example.order.dto.Order;
import com.example.order.service.BillingService;
import com.example.order.service.EmailService;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class OrderController {
    private final EmailService emailService;
    private final BillingService billingService;

    public OrderController(EmailService emailService, BillingService billingService) {
        this.emailService = emailService;
        this.billingService = billingService;
    }

    @PostMapping("/order")
    public void submitOrder(@RequestBody Order order) {

        billingService.charge(order);
        emailService.sendEmail(order);

    }

}

File: order-service-kotlin/src/main/kotlin/com/kotlin/spring/learning/controllers/OrderController.kt
package com.kotlin.spring.learning.controllers

import com.kotlin.spring.learning.config.RabbitAmqpConfig
import org.springframework.amqp.core.AmqpTemplate
import org.springframework.stereotype.Service
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

data class Order (var quantity: Int = 1, val title: String, val customerName: String)

@Service
class OrderService (var amqpTemplate: AmqpTemplate) {
    fun charge(order: Order) : Unit =   amqpTemplate.convertAndSend(RabbitAmqpConfig.DIRECT_ROUTING_KEY, order.toString())
}

@RestController
@RequestMapping("api")
class OrderController (var orderService: OrderService){

    @PostMapping("/order")
    fun submitOrder (@RequestBody order: Order) : Unit {

        orderService.charge(order)

    }
}